File: arena-backend/src/entities/direct-participant.entity.ts
import {
  Entity,
  JoinColumn,
  OneToOne,
  PrimaryColumn,
} from "typeorm";
import { ParticipantEntity } from "./participant.entity";

@Entity({ name: "direct_participants" })
export class DirectParticipantEntity {
  @PrimaryColumn({ type: "text" })
  channelId: string;

  @PrimaryColumn({ type: "text" })
  userId: string;

  @OneToOne(() => ParticipantEntity)
  @JoinColumn([
    { name: "channelId", referencedColumnName: "channelId" },
    { name: "userId", referencedColumnName: "userId" },
  ])
  participant: ParticipantEntity;
}

File: legacy/backend/src/main/java/app/sandori/arena/api/domain/channel/DirectParticipantEntity.java
package app.sandori.arena.api.domain.channel;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.IdClass;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "direct_participants")
@IdClass(ParticipantId.class)
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class DirectParticipantEntity {

    @Id
    private String channelId;

    @Id
    private String userId;

    public DirectParticipantEntity(String channelId, String userId) {
        this.channelId = channelId;
        this.userId = userId;
    }
}
